docs/apis/calcite.md

```diff
@@ -0,0 +1,195 @@
+
+    KafkaSinglePortInputOperator kafkaInput = dag.addOperator("KafkaInput", 
+                                                                KafkaSinglePortInputOperator.class);
+
+    kafkaInput.setInitialOffset("EARLIEST");
+
+    // Add CSVParser
+
+    CsvParser csvParser = dag.addOperator("CSVParser", CsvParser.class);
+
+    dag.addStream("KafkaToCSV", kafkaInput.outputPort, csvParser.in);
+
+Once we define DAG with Kafka InputOperator and CSVParser, it can parse data from Kafka topic. Till this point, this is a regular Apex application without using SQL. But with SQL on Apex, we can develop applications which uses hybrid DAG with part SQL and part regular apex implementation.
+
+The following code will describe how we can define StreamEndpoint. 
+ 
+    SQLExecEnvironment sqlEnv = sqlEnv.getEnvironment();
+
+    Map<String, Class> fieldMapping = ImmutableMap.<String, Class>of(
+
+       "RowTime", Date.class,
+
+       "id", Integer.class,
+
+       "Product", String.class,
+
+       "units", Integer.class);
+
+    sqlEnv = sqlEnv.registerTable("FROMCSV", new StreamEndpoint(csvParser.out, fieldMapping));
+
+To read existing data stream, we need to register it as a table with SQL execution environment with the name of the table and StreamEndpoint. StreamEndpoint can serve as input table or output table in SQL. For input table configuration we need to initialise StreamEndpoint with immediate upstream operator's output port and fieldMapping or POJO class for input tuple. For output table configuration, we need to initialise StreamEndpoint with immediate downstream operator's input port and fieldMapping or POJO class for output tuple. Once we register StreamEndpoint as a table with a name in SQL Execution Environment, it can be used as a table in SQL statement as other endpoints.
+
+Finally, we can define output table name and register it as a table with an environment before running SQL query.
+
+    // Register FileEndpoint as output table for  SQL.
+
+    sqlEnv = sqlEnv.registerTable("TOFILE", new FileEndpoint("/tmp/output",
+
+                        "out.file", new CSVMessageFormat(<<csvSchema>>)));
+
+    // Add SQL to DAG
+    sqlEnv.executeSQL(dag, sql);
+
+
+The UDF 'APEXCONCAT' is used in SQL as follows whereas other standard scalar SQL functions such as FLOOR, SUBSTRING are provided by Apache Calcite.
+
+    INSERT INTO TOFILE 
+           SELECT STREAM ROWTIME, FLOOR(ROWTIME TO DAY), APEXCONCAT('OILPAINT', SUBSTRING(PRODUCT, 6, 7)) 
+           FROM FROMCSV 
+           WHERE ID > 3 AND PRODUCT LIKE 'paint%'
+
+When executeSQL API is called, specified SQL is converted to DAG as described in the previous section. Both examples read CSV data from Kafka, but in pure style SQL example "KafkaInputOperator+ CSVParser" in DAG is created based on KafkaEndpoint usage while in fusion style example, it was already defined as part of DAG and same is extended with other operators as shown in the image below. 
```

Comment:
"Both examples read CSV data from Kafka. But in pure style SQL example "KafkaInputOperator+ CSVParser" in DAG is created based on KafkaEndpoint usage. While in fusion style example, it was already defined as part of DAG and same is extended with other operators as shown in the image below."


Correct sentence as above.